components/script/dom/eventtarget.rs

```diff
@@ -400,12 +407,25 @@ impl EventTarget {
                     entries.push(EventListenerEntry {
                         phase: ListenerPhase::Bubbling,
                         listener: EventListenerType::Inline(listener),
+                        once: false
                     });
                 }
             },
         }
     }
 
+    pub fn remove_listener_if_once(&self, ty: &Atom, listener: Rc<EventListener>) {
+        let mut handlers = self.handlers.borrow_mut();
+        let entry = handlers.get_mut(ty);
+
+        let listener = EventListenerType::Additive(listener.clone());
```

Comment:
Could you avoid the clone here, using a `match` expression in the closure passed to `Iterator::position`?

Reply:
Or I could just remove the clone and it's still work... No idea why it was there